src/main/kotlin/ar/edu/unahur/obj2/semillasAlViento/Parcela.kt

```diff
@@ -2,44 +2,41 @@ package ar.edu.unahur.obj2.semillasAlViento
 
 class Parcela(val ancho: Int, val largo: Int, val horasSolPorDia: Int) {
   val plantas = mutableListOf<Planta>()
-  var cantidadPlantas = 0
 
+
+  fun cantidadPlantas()= plantas.size
+  fun parcelaTieneComplicaciones(parcela: Parcela) =
+    parcela.plantas.any { it.horasDeSolQueTolera() < parcela.horasSolPorDia }
   fun superficie() = ancho * largo
   fun cantidadMaximaPlantas() =
-    if (ancho > largo) ancho * largo / 5 else ancho * largo / 3 + largo
+    if (ancho > largo) superficie() / 5 else superficie() / 3 + largo
 
   fun plantar(planta: Planta) {
-    if (cantidadPlantas == this.cantidadMaximaPlantas()) {
-      println("Ya no hay lugar en esta parcela")
+    if (plantas.size == this.cantidadMaximaPlantas()) {
+      throw error("LA PARCELA ESTA LLENA ")
     } else if (horasSolPorDia > planta.horasDeSolQueTolera() + 2) {
-      println("No se puede plantar esto acá, se va a quemar")
+      throw error("No se puede plantar esto acá, se va a quemar")
     } else {
       plantas.add(planta)
-      cantidadPlantas += 1
+
     }
   }
 }
 
 class Agricultora(val parcelas: MutableList<Parcela>) {
-  var ahorrosEnPesos = 20000
-
-  // Suponemos que una parcela vale 5000 pesos
-  fun comprarParcela(parcela: Parcela) {
-    if (ahorrosEnPesos >= 5000) {
-      parcelas.add(parcela)
-      ahorrosEnPesos -= 5000
-    }
-  }
 
   fun parcelasSemilleras() =
-    parcelas.filter {
-      parcela -> parcela.plantas.all {
+    parcelas.all { it.plantas.all {
         planta -> planta.daSemillas()
       }
```

Comment:
Faltó hacer este cambio. Cito la corrección anterior:

> El problema era de Acoplamiento: La Agricultora está acoplada a muchos detalles de implementación de la parcela que no debería conocer. Tendría que haber un método esSemillera en la Parcela

(o sea, este método tendría que ser `fun parcelasSemilleras() = parcelas.all { it.esSemillera() }` y el resto estar programado dentro de ese método)